packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/utils/check_version_compatibility.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectsType } from '@kbn/core-saved-objects-server';
+import {
+  getModelVersionsFromMappings,
+  compareModelVersions,
+  getModelVersionMapForTypes,
+  type IndexMapping,
+  type CompareModelVersionResult,
+} from '@kbn/core-saved-objects-base-server-internal';
+
+interface CheckVersionCompatibilityOpts {
+  mappings: IndexMapping;
+  types: SavedObjectsType[];
+  source: 'docVersions' | 'mappingVersions';
+  deletedTypes: string[];
+}
+
+export const checkVersionCompatibility = ({
+  mappings,
+  types,
+  source,
+  deletedTypes,
+}: CheckVersionCompatibilityOpts): CompareModelVersionResult => {
+  const appVersions = getModelVersionMapForTypes(types);
+  const indexVersions = getModelVersionsFromMappings({ mappings, source });
```

Comment:
Looking at code, this method will throw if any of the mappings versions is invalid. We don't seem to be catching this error here, nor in `init.ts` where we call `checkVersionCompatibility`. Is this the intended behavior?

Reply:
`getModelVersionMapForTypes` will never throw, because we validated each type during their registration.

`getModelVersionsFroMapping` could throw if someone manually tempered with the index / cluster state. But yeah in that case, it would be fatal. We could be more explicit instead of implicitly bubbling the thrown error though, eventually.